ext/semian/tickets.c

```diff
@@ -15,58 +21,75 @@ update_ticket_count(update_ticket_count_t *tc)
     }
 
     if (semctl(tc->sem_id, SI_SEM_CONFIGURED_TICKETS, SETVAL, tc->tickets) == -1) {
-      rb_raise(eInternal, "error updating max ticket count, errno: %d (%s)", errno, strerror(errno));
+      rb_raise(eInternal, "error configuring ticket count, errno: %d (%s)", errno, strerror(errno));
     }
   }
 
   return Qnil;
 }
-
 void
 configure_tickets(int sem_id, int tickets, int should_initialize)
 {
+  if (should_initialize) {
+    initialize_tickets(sem_id, tickets);
+  }
+
+  if (tickets > 0) {
+    configure_static_tickets(sem_id, tickets);
+  }
+}
+
+static void
+initialize_tickets(int sem_id, int tickets)
+{
   unsigned short init_vals[SI_NUM_SEMAPHORES];
+
+  if (tickets > 0) {
+    init_vals[SI_SEM_TICKETS] = init_vals[SI_SEM_CONFIGURED_TICKETS] = tickets;
```

Comment:
Why are you only initializing this if `tickets > 0`? With `init_vals` declared on the stack, these elements will contain garbage if not initialized here.